myhoard/backup_stream.py

```diff
@@ -149,7 +149,7 @@ def __init__(
             "mode": mode,
             "next_index": 0,
             "normalized_backup_time": normalized_backup_time,
-            "pending_binlogs": binlogs or [],
+            "pending_binlogs": list(binlogs) if binlogs else [],
```

Comment:
What are the binlog objects like, i.e. do we need deepcopy for this?

Reply:
Yes and no. The binlog objects themselves should not be modified so in that sense deepcopy would be required but this is already handled in `_upload_binlog`, which takes a copy of the `dict` itself before adding some fields.